Solaris specific patch, not suitable for upstream.

The pytest.ini changes are required as some of the
default options are not available in Solaris.

Additionally, a lot of test related packages are not available on Solaris
and are either patched out or corresponding tests skipped. Those are
'jaraco.context', 'jaraco.text', 'pypytools', 'pytest-mocker',
'requests_toolbelt', 'requests_unixsocket' and 'trustme'.

--- cheroot-11.1.2/pytest.ini
+++ cheroot-11.1.2/pytest.ini
@@ -1,8 +1,5 @@
 [pytest]
 addopts =
-  # `pytest-xdist`:
-  --numprocesses=auto
-
   # Show 10 slowest invocations:
   --durations=10
 
@@ -16,15 +13,6 @@ addopts =
   # https://docs.pytest.org/en/stable/doctest.html
   --doctest-modules
 
-  # Pre-load the `pytest-cov` plugin early:
-  -p pytest_cov
-
-  # `pytest-cov`:
-  --cov
-  --cov-config=.coveragerc
-  --cov-context=test
-  --no-cov-on-fail
-
   # Fail on config parsing warnings:
   # --strict-config
 
--- cheroot-11.1.2/cheroot/test/test_wsgi.py
+++ cheroot-11.1.2/cheroot/test/test_wsgi.py
@@ -7,8 +7,6 @@ import pytest
 
 import portend
 import requests
-from jaraco.context import ExceptionTrap
-from requests_toolbelt.sessions import BaseUrlSession as Session
 
 from cheroot import wsgi
 from cheroot._compat import IS_MACOS, IS_WINDOWS
@@ -38,7 +36,7 @@ def simple_wsgi_server():
         yield locals()
 
 
-@pytest.mark.flaky(reruns=3, reruns_delay=2)
+@pytest.mark.skip(reason="requests-toolbelt and jaraco.context is not available on Solaris")
 def test_connection_keepalive(simple_wsgi_server):
     """Test the connection keep-alive works (duh)."""
     session = Session(base_url=simple_wsgi_server['url'])
--- cheroot-11.1.2/cheroot/test/test_ssl.py
+++ cheroot-11.1.2/cheroot/test/test_ssl.py
@@ -16,7 +16,6 @@ import pytest
 
 import OpenSSL.SSL
 import requests
-import trustme
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives.serialization import (
     BestAvailableEncryption,
@@ -161,7 +160,7 @@ def tls_http_server(request):
 @pytest.fixture
 def ca():
     """Provide a certificate authority via fixture."""
-    return trustme.CA()
+    pytest.skip("trustme is not available on Solaris")
 
 
 @pytest.fixture
@@ -710,7 +709,6 @@ def test_https_over_http_error(http_serv
         pytest.param(ANY_INTERFACE_IPV6, marks=missing_ipv6),
     ),
 )
-@pytest.mark.flaky(reruns=3, reruns_delay=2)
 # pylint: disable-next=too-many-positional-arguments
 def test_http_over_https_error(
     http_request_timeout,
--- cheroot-11.1.2/cheroot/test/test_conn.py
+++ cheroot-11.1.2/cheroot/test/test_conn.py
@@ -13,8 +13,6 @@ from re import match as _matches_pattern
 
 import pytest
 
-from jaraco.text import trim, unwrap
-
 import cheroot.server
 from cheroot._compat import IS_CI, IS_MACOS, IS_PYPY, IS_WINDOWS
 from cheroot.test import helper, webtest
@@ -1357,13 +1355,11 @@ def test_No_Message_Body(test_client):
 
 
 @pytest.mark.xfail(
-    reason=unwrap(
-        trim("""
+    reason="""
         Headers from earlier request leak into the request
         line for a subsequent request, resulting in 400
         instead of 413. See cherrypy/cheroot#69 for details.
-        """),
-    ),
+        """
 )
 def test_Chunked_Encoding(test_client):
     """Test HTTP uploads with chunked transfer-encoding."""
--- cheroot-11.1.2/cheroot/test/conftest.py
+++ cheroot-11.1.2/cheroot/test/conftest.py
@@ -107,3 +107,8 @@ def make_http_server(bind_addr):
         time.sleep(0.1)
 
     return httpserver
+
+
+@pytest.fixture
+def mocker():
+    pytest.skip("pytest-mocker is not available on Solaris")
--- cheroot-11.1.2/cheroot/test/test_server.py
+++ cheroot-11.1.2/cheroot/test/test_server.py
@@ -16,8 +16,6 @@ from http import HTTPStatus
 import pytest
 
 import requests
-import requests_unixsocket
-from pypytools.gc.custom import DefaultGc
 
 from .._compat import IS_LINUX, IS_MACOS, IS_WINDOWS, SYS_PLATFORM, bton, ntob
 from ..server import IS_UID_GID_RESOLVABLE, Gateway, HTTPServer
@@ -267,7 +265,7 @@ def peercreds_enabled_server(http_server
 
 @unix_only_sock_test
 @non_macos_sock_test
-@pytest.mark.flaky(reruns=3, reruns_delay=2)
+@pytest.mark.skip(reason="requests_unixsocket is not available on Solaris")
 def test_peercreds_unix_sock(http_request_timeout, peercreds_enabled_server):
     """Check that ``PEERCRED`` lookup works when enabled."""
     httpserver = peercreds_enabled_server
@@ -298,6 +296,7 @@ def test_peercreds_unix_sock(http_reques
         assert peercreds_text_resp.status_code == 500
 
 
+@pytest.mark.skip(reason="requests_unixsocket is not available on Solaris")
 @pytest.mark.skipif(
     not IS_UID_GID_RESOLVABLE,
     reason='Modules `grp` and `pwd` are not available '
@@ -418,8 +417,7 @@ def test_reuse_port(http_server, ip_addr
 @pytest.fixture
 def _garbage_bin():
     """Disable garbage collection when this fixture is in use."""
-    with DefaultGc().nogc():
-        yield
+    pytest.skip("pypytools is not available on Solaris")
 
 
 @pytest.fixture
